server/container/guice/mailet/src/test/java/org/apache/james/utils/GuilceMailetLoaderTest.java

```diff
@@ -0,0 +1,163 @@
+        }
+
+        @Override
+        public File getFile(String fileURL) throws FileNotFoundException {
+            return new File(ClassLoader.getSystemResource("recursive/included-jars").getFile());
+        }
+
+        @Override
+        public File getBasedir() throws FileNotFoundException {
+            throw new NotImplementedException();
+        }
+    };
+
+    public static final FileSystem RECURSIVE_CLASSPATH_FILE_SYSTEM = new FileSystem() {
+        @Override
+        public InputStream getResource(String url) throws IOException {
+            throw new NotImplementedException();
+        }
+
+        @Override
+        public File getFile(String fileURL) throws FileNotFoundException {
+            return new File(ClassLoader.getSystemResource("recursive/").getFile());
+        }
+
+        @Override
+        public File getBasedir() throws FileNotFoundException {
+            throw new NotImplementedException();
+        }
+    };
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+    private Injector injector = Guice.createInjector();
+
+    @Test
+    public void getMailetShouldLoadClass() throws Exception {
+        GuiceMailetLoader guiceMailetLoader = new GuiceMailetLoader(injector, THROWING_FILE_SYSTEM);
+
+        Mailet mailet = guiceMailetLoader.getMailet(FakeMailetConfig.builder()
+            .mailetName("AddFooter")
+            .mailetContext(FakeMailContext.defaultContext())
+            .build());
+
+        assertThat(mailet).isInstanceOf(AddFooter.class);
+    }
+
+    @Test
+    public void getMailetShouldThrowOnBadType() throws Exception {
+        GuiceMailetLoader guiceMailetLoader = new GuiceMailetLoader(injector, THROWING_FILE_SYSTEM);
+
+        expectedException.expect(MessagingException.class);
+
+        guiceMailetLoader.getMailet(FakeMailetConfig.builder()
+            .mailetName("org.apache.james.transport.matchers.SizeGreaterThan")
+            .mailetContext(FakeMailContext.defaultContext())
+            .build());
+    }
+
+    @Test
+    public void getMailetShouldLoadClassWhenInIncludedJars() throws Exception {
+        GuiceMailetLoader guiceMailetLoader = new GuiceMailetLoader(Guice.createInjector(), CLASSPATH_FILE_SYSTEM);
+
+        Mailet mailet = guiceMailetLoader.getMailet(FakeMailetConfig.builder()
+            .mailetName("org.apache.transport.mailets.AIMailet")
+            .mailetContext(FakeMailContext.defaultContext())
+            .build());
+
+        assertThat(mailet.getClass().getCanonicalName())
+            .isEqualTo("org.apache.transport.mailets.AIMailet");
+    }
+
+    @Test
+    public void getMailedShouldShouldRecursivelyIncludeJar() throws Exception {
+        GuiceMailetLoader guiceMailetLoader = new GuiceMailetLoader(Guice.createInjector(), RECURSIVE_CLASSPATH_FILE_SYSTEM);
+
+        Mailet mailet = guiceMailetLoader.getMailet(FakeMailetConfig.builder()
+            .mailetName("org.apache.transport.mailets.AIMailet")
```

Comment:
can you add a test with a mailet in another package such as com.example.mailets.CustomMailet?